### DISCUSSION TITLE
Compatible with PHP versions 7.4? -> error

### DISCUSSION THREAD

**Discussion by philippehenri** [tectalichq/public-openai-client-php#29] [Q&A] [answered]
Hi

I read that "Compatible with PHP versions 7.2 → 8.1[¶](https://tectalic.com/blog/openai-php-client-library-sdk#compatible-with-php-versions-72--81)"

but when I try :   composer require tectalic/openai

I get the below error, requiring php 8??

What am I doing wrong ( running on debian 11 , php 7.2 )?
thanks 
philippe

..../composer.json has been updated
Running composer update tectalic/openai
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - openai-php/client is locked to version v0.5.2 and an update of this package was not requested.
    - openai-php/client v0.5.2 requires php ^8.1.0 -> your php version (7.4.33) does not satisfy that requirement.
  Problem 2
    - symfony/http-client is locked to version v6.2.11 and an update of this package was not requested.
    - symfony/http-client v6.2.11 requires php >=8.1 -> your php version (7.4.33) does not satisfy that requirement.

You can also try re-running composer require with an explicit version constraint, e.g. "composer require tectalic/openai:*" to figure out if any version is installable, or "composer require tectalic/openai:^2.1" if you know which you need.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

**Comment by the-csaba** [comment, marked as answer]
Hi @philippehenri ,

It appears you are having trouble installing the Tectalic OpenAI PHP SDK with Composer due to PHP version compatibility issues. According to your error log, your current PHP version is 7.4.33, compatible with the Tectalic OpenAI PHP SDK. 

However, it seems like there might be confusion between two different packages. 

Please try to install `tectalic/openai`, which should be compatible with your PHP version.

Here's what you can do:

1. Run `composer clear-cache` to clear your composer cache.
2. Ensure that `openai-php/client` is not listed in your `composer.json` file; if it is, remove it.
3. Run `composer update` to ensure all your existing dependencies are up to date.
4. Finally, try running `composer require tectalic/openai` again.

If you continue to have issues, you might want to consider updating your PHP version to meet the requirements of all your dependencies. 

Best,

Csaba
